src/vs/workbench/contrib/debug/node/debugAdapter.ts

```diff
@@ -136,6 +136,51 @@ export class SocketDebugAdapter extends StreamDebugAdapter {
 	}
 }
 
+/**
+ * An implementation that connects to a debug adapter via a NamedPipe (on Windows)/UNIX Domain Socket (on non-Windows).
+ */
+export class NamedPipeDebugAdapter extends StreamDebugAdapter {
+
+	private socket?: net.Socket;
+
+	constructor(private adapterServer: IDebugAdapterNamedPipeServer) {
+		super();
+	}
+
+	startSession(): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			let connected = false;
+			this.socket = net.createConnection(this.adapterServer.path, () => {
```

Comment:
Since the logic is otherwise identical to `SocketDebugAdapter`, I would probably pull this logic into an `abstract class NetworkDebugAdapter` that has an abstract `createConnection` method that the Socket and NamedPipe debug adapters can implement.

Reply:
Good call - refactored that to a `NetworkDebugAdapter` abstract class.

Reply:
I meant to also move the rest of the logic within `startSession()` to the parent class, since the only thing different is the arg to `net.createConnection` 🙂

Reply:
Oh yeah of course :) now I have an abstract `createConnection` that `SocketDebugAdapter` and `NamedPipeDebugAdapter` implement.